llvm/lib/Transforms/Utils/CodeLayout.cpp

```diff
@@ -516,34 +518,62 @@ struct MergedChain {
   NodeIter End3;
 };
 
+/// A wrapper around two concatenated vectors (chains) of jumps.
+struct MergedJumpsT {
+  using JumpIter = std::vector<JumpT *>::const_iterator;
+
+  MergedJumpsT(JumpIter Begin, JumpIter End) : Begin1(Begin), End1(End) {}
+
+  template <typename F> void forEach(const F &Func) const {
+    for (auto It = Begin1; It != End1; It++)
+      Func(*It);
+    for (auto It = Begin2; It != End2; It++)
+      Func(*It);
+  }
+
+  bool empty() const { return Begin1 == End1; }
+
+  void append(JumpIter Begin, JumpIter End) {
+    assert(Begin2 == End2 && "cannot extend MergedJumpsT");
+    Begin2 = Begin;
+    End2 = End;
+  }
+
+private:
+  JumpIter Begin1;
```

Comment:
Are `Begin1` and `End1` always going to be the beginning and ending of a single vector? If yes, can we simplify this further by storing pointers to two vectors:
`std::array<std::vector<JumpT *>, 2> Jumps;`

Reply:
Wouldn't `std::array<std::vector<JumpT *>, 2>` instantiate the vector during construction? That's exactly what I want to avoid. Or did you mean `std::array<std::pair<JumpIter, JumpIter>, 2>`?

Reply:
Ah, I actually meant `std::array<std::vector<JumpT *>*, 2>`.

Reply:
Changed to `std::array<std::vector<JumpT *>*, 2>` as suggested. 
I really hope that you find the new version cleaner/simpler because I am not sure about myself :)